src/org/zaproxy/zap/authentication/PostBasedAuthenticationMethodType.java

```diff
@@ -236,6 +243,7 @@ protected HttpMessage prepareRequestMessage(UsernamePasswordAuthenticationCreden
 				if (requestBody != null) {
 					requestMessage.getRequestBody().setBody(requestBody);
 					requestMessage.getRequestHeader().setHeader(HttpHeader.CONTENT_LENGTH, null);
+					replaceAntiCsrfTokenAndCookieValueIfRequired(requestMessage);
```

Comment:
This should be done also when not using a `SiteNode`. Any reason not to do this in `authenticate` instead?

Reply:
I am replacing the old token value with the new one. To get the old one I need the history reference to the very first post request to login site.

Reply:
Whether the authentication was set up with a `SiteNode` or manually does not matter for the token generation (`AntiCsrfToken` already has the message to regenerate the token).

Reply:
then, can I rely on `loginRequestBody` and `loginRequestURL` in `PostBasedAuthenticationMethod` to get the `AntiCsrfToken` using `extAntiCsrf.getTokens(reqBody, reqUri)`?

Reply:
Yes, if `loginRequestBody` and `loginRequestURL` have the expected contents it should work.

Reply:
The problem is that we don't have any guarantees that's still fresh/valid (specially if using another session), so the authentication would fail and we would have to retry it after getting a valid one.